### PR TITLE
#1034 project edit user select auto complete

### LINKED ISSUE #1034: Project Edit - Make user selects be autocompletes
### Description

When you select a project manager or project lead a massive list appears to select from. this should be an autocomplete instead with the same people listed already. 

### Acceptance Criteria

Is an autocomplete

### Proposed Solution

Change this part of the form to be NER autocompletes. See Either role management, change request creation. 

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by ryanhowe28**
## Changes

Made the drop downs into autocompletes

## Notes

doesn't successfully edit the project manager or lead

## Screenshots
image

image

## To Do

_Any remaining things that need to get done_

- [ ] get it to successfully change

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #1034

### CODE REVIEW

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx [hunk 1]
```diff
@@ -13,6 +14,45 @@ interface ProjectEditDetailsProps {
 }
 
 const ProjectEditDetails: React.FC<ProjectEditDetailsProps> = ({ users, control, errors }) => {
+  const [projectLead, setProjectLead] = useState<User | null>(null);
+  const [projectManager, setProjectManager] = useState<User | null>(null);
```
Comment (ryanhowe28) [inline src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx:18]: shouldn't be null, should start with the current project lead or manager

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditContainer.tsx [hunk 2]
```diff
@@ -90,46 +91,36 @@ const ProjectEditContainer: React.FC<ProjectEditContainerProps> = ({ project, ex
...
+        otherConstraints,
+        projectLeadId: projectLeadId ? parseInt(projectLeadId) : undefined,
+        projectManagerId: projectManagerId ? parseInt(projectManagerId) : undefined
+      };
       await mutateAsync(payload);
       exitEditMode();
     } catch (e) {
       if (e instanceof Error) {
         toast.error(e.message);
+        return;
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditContainer.tsx:123]: i dont think you need this

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx [hunk 3]
```diff
@@ -1,18 +1,36 @@
...
+import NERAutocomplete from '../../../components/NERAutocomplete';
+import { Dispatch, SetStateAction } from 'react';
 
 interface ProjectEditDetailsProps {
   users: User[];
   control: any;
   errors: any;
+  projectManager: string | undefined;
+  projectLead: string | undefined;
+  setProjectManager: Dispatch<SetStateAction<string | undefined>>;
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx:16]: reid told me to change this to (projectManager?: string) => void

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx [hunk 4]
```diff
@@ -1,18 +1,36 @@
...
+  setProjectLead: Dispatch<SetStateAction<string | undefined>>;
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx:17]: same thing here

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx [hunk 5]
```diff
@@ -40,47 +58,27 @@ const ProjectEditDetails: React.FC<ProjectEditDetailsProps> = ({ users, control,
...
-                        {fullNamePipe(t)}
-                      </MenuItem>
-                    ))}
-                  </TextField>
-                </>
-              )}
-            />
-          </FormControl>
+        <Grid item xs={12} md={3}>
+          <FormLabel>Project Lead</FormLabel>
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx:62]: reid also said to get rid of this cause its redundant with the placeholder

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx [hunk 6]
```diff
@@ -40,47 +58,27 @@ const ProjectEditDetails: React.FC<ProjectEditDetailsProps> = ({ users, control,
...
-                        {fullNamePipe(t)}
-                      </MenuItem>
-                    ))}
-                  </TextField>
-                </>
-              )}
-            />
-          </FormControl>
+        <Grid item xs={12} md={3}>
+          <FormLabel>Project Manager</FormLabel>
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx:73]: same thing here

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
Looks Great! nitpicks + merge conflicts😭

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx [hunk 7]
```diff
@@ -1,19 +1,35 @@
...
 import AttachMoneyIcon from '@mui/icons-material/AttachMoney';
 import { fullNamePipe } from '../../../utils/pipes';
-import { ProjectEditFormInput } from './ProjectEditContainer';
+import NERAutocomplete from '../../../components/NERAutocomplete';
 
 interface ProjectEditDetailsProps {
   users: User[];
-  control: Control<ProjectEditFormInput>;
-  errors: FieldErrorsImpl<ProjectEditFormInput>;
+  control: any;
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx:11]: change these types back to what they were before (look on github) I did the same thing idk why

**Review by Peyton-McKee — DISMISSED** [review]
LGTM!

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx [hunk 8]
```diff
@@ -1,19 +1,37 @@
...
+const ProjectEditDetails: React.FC<ProjectEditDetailsProps> = ({
+  users,
+  control,
+  errors,
+  projectManager,
+  projectLead,
+  setProjectLead,
+  setProjectManager
+}) => {
+  const userToAutocompleteOption = (user?: User): { label: string; id: string } => {
```
Comment (RChandler234) [inline src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx:30]: move this method above the component since it's just acting as a helper

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx [hunk 9]
```diff
@@ -1,19 +1,37 @@
...
 import { fullNamePipe } from '../../../utils/pipes';
+import NERAutocomplete from '../../../components/NERAutocomplete';
 import { ProjectEditFormInput } from './ProjectEditContainer';
+import { Control, FieldErrorsImpl } from 'react-hook-form';
 
 interface ProjectEditDetailsProps {
   users: User[];
   control: Control<ProjectEditFormInput>;
   errors: FieldErrorsImpl<ProjectEditFormInput>;
+  projectManager: string | undefined;
```
Comment (RChandler234) [inline src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx:15]: projectManager?: string (same for projectLead)

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditContainer.tsx [hunk 10]
```diff
@@ -111,50 +108,41 @@ const ProjectEditContainer: React.FC<ProjectEditContainerProps> = ({ project, ex
...
-      features,
-      otherConstraints,
-      projectLeadId,
-      projectManagerId
-    };
-
     try {
+      const payload = {
+        name,
+        budget: parseInt(budget),
```
Comment (RChandler234) [inline src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditContainer.tsx:131]: why did you change this line?

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditContainer.tsx [hunk 11]
```diff
@@ -111,50 +108,41 @@ const ProjectEditContainer: React.FC<ProjectEditContainerProps> = ({ project, ex
...
 
   if (allUsers.isLoading || !allUsers.data) return <LoadingIndicator />;
   if (allUsers.isError) {
     return <ErrorPage message={allUsers.error?.message} />;
   }
 
   const users = allUsers.data.filter((u) => u.role !== 'GUEST');
+  const onSubmit = async (data: any) => {
+    const { name, budget, summary, bomLink, googleDriveFolderLink, taskListLink, slideDeckLink } = data;
+    const rules = data.rules.map((rule: any) => rule.rule || rule);
```
Comment (RChandler234) [inline src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditContainer.tsx:122]: why did you change lines 121 and 122, don't think you need to change them for this ticket

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx [hunk 12]
```diff
@@ -41,47 +59,25 @@ const ProjectEditDetails: React.FC<ProjectEditDetailsProps> = ({ users, control,
...
-          </FormControl>
+        <Grid item xs={12} md={3}>
+          <NERAutocomplete
+            id="users-autocomplete"
+            onChange={(_event, value) => setProjectLead(value?.id)}
+            options={users.map(userToAutocompleteOption)}
+            size="small"
+            placeholder="Select a Project Lead"
+            value={userToAutocompleteOption(users.find((user) => user.userId.toString() === projectLead))}
+          />
```
[the remark this answers is not in the document]
Comment (Peyton-McKee) [reply src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx:70]: you told me to get rid of my form labels.

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx (hunk 8 above)
Comment (Peyton-McKee) [reply src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditDetails.tsx:30]: we use this method so often we should extract it to a different file, export it and use the same one everywhere

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditContainer.tsx (hunk 11 above)
Comment (ryanhowe28) [reply src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditContainer.tsx:122]: Line 122 wasn't changed, but I changed line 121 back

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditContainer.tsx (hunk 10 above)
Comment (ryanhowe28) [reply src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditContainer.tsx:131]: I saw what we did with the crId and thought it would be a good idea

File: src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditContainer.tsx (hunk 11 above)
Comment (ryanhowe28) [reply src/frontend/src/pages/ProjectDetailPage/ProjectEdit/ProjectEditContainer.tsx:122]: Also we set the value in the payload, so its redundant?

**Comment by RChandler234** [comment]
You removed the replacement for an any type, I just added that change and then merged it cause I wanted to wrap this one up
